tests/core-addon/hooks.js

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var browser = require('sinon-chrome/extensions');
+// We need to provide the `browser.runtime.id` for sinon-chrome to
+// be happy and play nice with webextension-polyfill. See this issue:
+// https://github.com/mozilla/webextension-polyfill/issues/218
+browser.runtime.id = "testid";
+global.browser = browser;
```

Comment:
This was required, apparently the `beforeAll` hook was not sufficient.

Reply:
In this case, there is no need for the `beforeAll` hook?

Reply:
It's still required. We want to flush/remove in `afterAll` and then reinstate in `beforeAll`. I suspect the first `global.browser` is used way earlier than running the tests!